indexer/src/main/java/au/org/aodn/esindexer/utils/MapperUtils.java

```diff
@@ -473,21 +478,5 @@ public static List<ContactsModel> addRoleToContacts(List<ContactsModel> contacts
         return contacts;
     }
 
-    /**
-     * This function is used to get rid of too much null checking.
-     * Example:
-     * For getting a deep value like this:
-     * <code>phoneCode = phone.getCITelephone().getNumberType().getCITelephoneTypeCode();</code>
-     * every getter needs to check null.
-     * @param supplier The function that may throw NullPointerException
-     * @param <T>      The type of the return value
-     * @return null if any of the getter is null
-     */
-    public static <T> Optional<T> safeGet(Supplier<T> supplier) {
-        try {
-            return Optional.of(supplier.get());
-        } catch (NullPointerException ignored) {
-            return Optional.empty();
-        }
-    }
```

Comment:
so you created new class class `SafeGetUtils` for only one single public `safeGet` method, I understand that `safeGet` can be more high level than the scope of `MapperUtils` methods, which why you wanted to separate them, but I feel it is not necessary to have single utils class to serve one single method, maybe you should consider restructure/rename the MapperUtils class into something more generic, then all utils methods can be in there... OR rename SafeGetUtils to something more generic, so future misc utils methods can be chucked into, not just the safeGet. Otherwise, every time you have new utils method, you will create one more new classes. The utils class is not the model class, no need to break into very small scope.

Reply:
Thank you. Considering another Util is mapperUtils, what about naming it GetterUtils? Or do you think it should still be in the MapperUtils rather than a new class?

Reply:
it is ok to have separated utils classes if they serve distinct purposes, e.g date utils, string utils, mapper utils, gn4 utils, whatever ; if the purposes is so unique that it has to be in a separated utils class, and that class only has just one public method, name it singularly e.g `SomethingUniqueUtil` rather than `SomethingUniqueUtils`. 

given the safeGet can be for anything that needs safely handling null pointer exception, any other utils classes might consume safeGet, might be have a generic utils class called `CommonUtils` to store any high level utils methods? or if safeGet only be used for stuff related to Mapping , then might be move back to `MapperUtils`. Up to you to decide, just not put it in`SafeGetUtils`.

Reply:
it is odd like you used to name safeGet as getNullifNull in the past, too specific